lib/Reader/readerir.cpp

```diff
@@ -742,6 +744,51 @@ void GenIR::createSym(uint32_t Num, bool IsAuto, CorInfoType CorType,
   }
 }
 
+void GenIR::zeroInitLocals() {
+  bool InitAllLocals = isZeroInitLocals();
+  for (const auto &LocalVar : LocalVars) {
+    Type *LocalTy = LocalVar->getType()->getPointerElementType();
+    if (InitAllLocals || isManagedType(LocalTy)) {
+      // TODO: if InitAllLocals is false we only have to zero initialize
+      // GC pointers and GC pointer fields on structs. For now we are zero
+      // initalizing all fields in structs that have gc fields.
+      // TODO: We should try to lay out GC pointers contiguously on the stack
+      // frame and use memset to initialize them.
+      // TODO: We can avoid zero-initializing some gc pointers if we can
+      // ensure that we are not reporting uninitialized GC pointers at gc-safe
+      // points.
+      StructType *StructTy = dyn_cast<StructType>(LocalTy);
+      if (StructTy != nullptr) {
+        const DataLayout *DataLayout = JitContext->EE->getDataLayout();
+        const StructLayout *TheStructLayout =
+            DataLayout->getStructLayout(StructTy);
+        zeroInitBlock(LocalVar, TheStructLayout->getSizeInBytes());
+      } else {
+        Constant *ZeroConst = Constant::getNullValue(LocalTy);
+        LLVMBuilder->CreateStore(ZeroConst, LocalVar);
+      }
+    }
+  }
+}
+
+void GenIR::zeroInitBlock(Value *Address, uint64_t Size) {
+  bool IsSigned = false;
+  ConstantInt *BlockSize = ConstantInt::get(
+      *JitContext->LLVMContext, APInt(TargetPointerSizeInBits, Size, IsSigned));
+  zeroInitBlock(Address, BlockSize);
+}
+
+void GenIR::zeroInitBlock(Value *Address, Value *Size) {
+  // TODO: For small structs we may want to generate an integer StoreInst
+  // instead of calling a helper.
+  LLVMContext &LLVMContext = *JitContext->LLVMContext;
+  const bool MayThrow = false;
+  Type *VoidTy = Type::getVoidTy(LLVMContext);
+  Value *ZeroByte = ConstantInt::get(LLVMContext, APInt(8, 0, true));
+  callHelperImpl(CORINFO_HELP_MEMSET, MayThrow, VoidTy, (IRNode *)Address,
```

Comment:
Yes. From #417: "Getting the LLVM IR right here is not too hard. Note we're zeroing things via the helper since using LLVM's memset might lead to calls into the CRT." @AndyAyersMS Is there an issue on this already?